python/src/file_io.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABCMeta, abstractmethod
```

Comment:
What is this going to be used for? Just general writing to files?

Reply:
Yup. Defining a File interface means we can make a mock version for tests so tests don't actually have to write to the file system